doc/nrf/ug_bootloader.rst

```diff
@@ -92,7 +92,96 @@ These signed variants can be used to perform an upgrade procedure through the :r
+Both bootloaders can easily be included in your application using :ref:`ug_multi_image`.
+
+Adding the immutable bootloader
+==============================
+
+To add an immutable bootloader to your application, set :option:`CONFIG_SECURE_BOOT` and add your private key file under :option:`CONFIG_SB_SIGNING_PUBLIC_KEY`.
+|how_to_configure|
+
+See the documentation of the :ref:`bootloader` sample for more information.
+The :ref:`bootloader_build_and_run` section has detailed instructions for adding the immutable bootloader as first stage of the secure bootloader chain.
+
+Adding MCUboot as an upgradable bootloader
+===============================
+
+To add an upgradable bootloader in this case MCUBoot to your applications, set :option:`CONFIG_BOOTLOADER_MCUBOOT`.
+|how_to_configure|
+
+If you want an upgradable bootloader you should always include the immutable bootloader as well when adding MCUBoot as an upgradable bootloader.
+
+.. note::
+   It is possible to include this bootloader without the immutable bootloader.
+   In this case MCUboot will act as an immutable bootloader.
+
+   In the default configuration, most |NCS| samples include the upgradable bootloader but not the immutable bootloader.
+   The reason for this is that the immutable bootloader must be configured with a private key before use.
+
+The |NCS| uses MCUboot as upgradable bootloader.
+See :doc:`mcuboot:index` for information about the default implementation.
 :ref:`mcuboot:mcuboot_ncs` gives details on the integration of MCUboot in |NCS|.
+
+You can configure MCUboot by setting configuration options for the ``mcuboot`` child image.
+
+Flash partitions used by MCUboot
+--------------------------------
+
+MCUboot requires two image slots: one that contains the application to be booted (the *primary slot*), and one where a new application can be stored before it is activated (the *secondary slot*).
+See the *Image Slots* section in the :doc:`MCUboot documentation <mcuboot:design>` for more information.
+
+The |NCS| variant of MCUboot uses the :ref:`partition_manager` to configure the flash partitions for these image slots.
+
+In the default configuration, the Partition Manager dynamically sets up the partitions as required.
+If you want to control where in memory the flash partitions are placed, you can define static partitions for your application.
+See :ref:`ug_pm_static` for more information.
+
+It is possible to use external flash as the storage partition for the secondary slot.
+This requires a driver for the external flash that supports:
+
+* Single-byte read and write
+* Writing data from internal flash to external flash
+
+See :ref:`pm_external_flash` for general information about how to set up partitions in external flash in the Partition Manager.
+To configure MCUboot to use external flash for the secondary slot, edit the :file:`ncs/bootloader/mcuboot/boot/zephyr/pm.yml` with the following changes file::
```

Comment:
"the following changes file"?! Do you mean:
```suggestion
To configure MCUboot to use external flash for the secondary slot, edit the :file:`ncs/bootloader/mcuboot/boot/zephyr/pm.yml` file with the following changes::
```
I don't like that wording. Maybe:
```suggestion
To configure MCUboot to use external flash for the secondary slot, update the :file:`ncs/bootloader/mcuboot/boot/zephyr/pm.yml` file to contain the following definition for ``mcuboot_secondary``::
```